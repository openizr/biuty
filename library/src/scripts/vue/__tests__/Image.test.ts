/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UIImage from 'scripts/vue/Image.vue';

describe('vue/UIImage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UIImage, {
      propsData: {
        alt: 'test', ratio: 'square', src: 'https://test.com/a.jpg', modifiers: 'large',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - custom ratio', () => {
    const wrapper = mount(UIImage, {
      propsData: { alt: 'test', ratio: '25:32', src: 'https://test.com/a.jpg' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UIImage, {
      propsData: {
        id: 'test', alt: 'test', ratio: 'portrait', src: 'https://test.com/a.jpg',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - standard ratio with itemprop', () => {
    const wrapper = mount(UIImage, {
      propsData: {
        itemProp: 'image', alt: 'test', ratio: 'landscape', src: 'https://test.com/a.jpg',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - custom ratio with itemprop', () => {
    const wrapper = mount(UIImage, {
      propsData: {
        itemProp: 'image', alt: 'test', ratio: '1:5', src: 'https://test.com/a.jpg',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with panoramic ratio', () => {
    const wrapper = mount(UIImage, {
      propsData: { alt: 'test', ratio: 'panoramic', src: 'https://test.com/a.jpg' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
