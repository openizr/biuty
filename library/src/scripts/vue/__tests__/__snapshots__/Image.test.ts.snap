// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/UIImage renders correctly - basic 1`] = `"<div class=\\"ui-image ui-image--square--large\\"><img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" width=\\"1\\" height=\\"1\\"></div>"`;

exports[`vue/UIImage renders correctly - custom ratio 1`] = `"<img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" width=\\"25\\" height=\\"32\\" class=\\"ui-image ui-image--25x32\\">"`;

exports[`vue/UIImage renders correctly - custom ratio with itemprop 1`] = `"<img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" width=\\"1\\" height=\\"5\\" itemprop=\\"image\\" class=\\"ui-image ui-image--1x5\\">"`;

exports[`vue/UIImage renders correctly - standard ratio with itemprop 1`] = `"<div class=\\"ui-image ui-image--landscape\\"><img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" itemprop=\\"image\\" width=\\"3\\" height=\\"2\\"></div>"`;

exports[`vue/UIImage renders correctly - with id 1`] = `"<div class=\\"ui-image ui-image--portrait\\"><img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" width=\\"2\\" height=\\"3\\"></div>"`;

exports[`vue/UIImage renders correctly - with panoramic ratio 1`] = `"<div class=\\"ui-image ui-image--panoramic\\"><img src=\\"https://test.com/a.jpg\\" alt=\\"test\\" loading=\\"lazy\\" width=\\"16\\" height=\\"9\\"></div>"`;
